File: IESBuscador/WebContent/js/service/personService.js
app.factory("personService", function($q, $http) {
	return {
		addPerson: function(person){			
			return $http.put('http://localhost:8080/IESBuscador/rest/personServices/addPerson/', person).then(function(data){
				var received=angular.fromJson(data);				
				return received;
			}).catch(function(data){
				console.log(data);
			});
		},
		
		findByEmail: function(email){
			return $http.get("http://localhost:8080/IESBuscador/rest/personServices/findByEmail/"+email).then(function(data){
				var received=angular.fromJson(data);
				return received;
			}).catch(function(data){
				console.log(data);
			});
		},
		
		updatePerson: function(personUpdated){
			return $http.put("http://localhost:8080/IESBuscador/rest/personServices/updatePerson/",personUpdated).then(function(data){
				var received=angular.fromJson(data);
				return received;
			}).catch(function(data){
				console.log(data);
			});
		}
	}
});

File: IESBuscador/src/rest/PersonService.java
package rest;

import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import bussiness.person.IPersonBussiness;
import bussiness.person.impl.PersonBussiness;
import dto.PersonDTO;

@Path("/personServices")
public class PersonService {
	
	private IPersonBussiness personBussiness;
	
	public PersonService(){
		personBussiness = new PersonBussiness();
	}
	
	@PUT
	@Path("/addPerson")
	@Consumes (MediaType.APPLICATION_JSON)
	@Produces (MediaType.TEXT_PLAIN)
	public String addPerson(PersonDTO person){
		return personBussiness.addPerson(person);
	}
	
	@GET
	@Path("/findByEmail/{email}")
	@Produces(MediaType.APPLICATION_JSON)
	public PersonDTO findByEmail(@PathParam("email") String email) {
		return personBussiness.findByEmail(email);
	}
	
	@PUT
	@Path("/updatePerson")
	@Consumes (MediaType.APPLICATION_JSON)
	@Produces (MediaType.TEXT_PLAIN)
	public String updatePerson(PersonDTO person){
		return personBussiness.updatePerson(person);
	}
}
